lib-multisrc/iken/src/eu/kanade/tachiyomi/multisrc/iken/Iken.kt

```diff
@@ -9,25 +14,33 @@ import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import eu.kanade.tachiyomi.source.online.HttpSource
 import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.Serializable
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import okhttp3.HttpUrl.Companion.toHttpUrl
 import okhttp3.Request
 import okhttp3.Response
+import org.jsoup.nodes.Document
 import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import uy.kohesive.injekt.injectLazy
 
 abstract class Iken(
     override val name: String,
     override val lang: String,
     override val baseUrl: String,
-) : HttpSource() {
+) : HttpSource(), ConfigurableSource {
 
     override val supportsLatest = true
 
     override val client = network.cloudflareClient
 
-    private val json by injectLazy<Json>()
+    protected val json by injectLazy<Json>()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
```

Comment:
```suggestion
    private val preferences: SharedPreferences by getPreferencesLazy()
```
need to import as well